File: 442-find-all-duplicates-in-an-array/442.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        List<Integer> result = new ArrayList<Integer>();
        
        for (int x : nums) {
            x = Math.abs(x);
            if (nums[x - 1] < 0) {
                result.add(x);
            } else {
                nums[x - 1] *= -1;
            }
        }

        return result;
    }
}

File: my-solutions/442-find-all-duplicates-in-an-array/442.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDuplicates = function(nums) {
    const result = [];
    for (let i = 0; i < nums.length; i++) {
        const cur_num = Math.abs(nums[i])
        if (nums[cur_num - 1] < 0) {
            result.push(cur_num)
        } else {
            nums[cur_num - 1] *= -1
        }
    }

    return result
};
